### DISCUSSION TITLE
Brightness in i3status

### DISCUSSION THREAD

**Discussion by kora24** [i3/i3#6035] [Q&A] [answered]
Hi everyone,

I installed _xorg-server, xorg-xbacklight, xf86-video-intel_;

then added
`bindsym XF86MonBrightnessUp exec --no-startup-id "xbacklight -inc 10"`  
`bindsym XF86MonBrightnessDown exec --no-startup-id "xbacklight -dec 10"`
to the _~/.config/i3/config_

The brightness keys work perfectly. The `$ xbacklight -get` shows whatever it should.

I am trying now to add the brightness to the _i3status.conf_ . I suppose I need to add
`order += "brightness"` and
`brightness { format = "..." }`
but what exactly I cannot figure out.

Please advise.

Thank you!

**Comment by okraits** [comment, marked as answer]
Did you read the i3status mapage? i3status doesn't support displaying the monitor brightness.
But you can display the content of a file: https://man.archlinux.org/man/i3status.1#File_Contents
So you can do:

1. write a script which periodically writes the output of `xbacklight -get` to a temporary file
2. execute that script from your i3 config
3. configure i3status to read and display the content of that temporary file

**Reply by pflingstring** [reply]
Alternatively, one could read the [brightness level directly from the kernel](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/ABI/stable/sysfs-class-backlight#n12) and skip steps 1 and 2.

**Reply by okraits** [reply]
Yes, although you might not get a nice relative value like `40 %` but rather an absolute value like `7680`.

**Comment by kora24** [comment]
Thank you very much!
Yes, I read the i3status mapage but thought I'd ask anyway. It's working now.
Thanks again!
